test/livebook/live_markdown/markdown_helpers_test.exs

```diff
@@ -243,6 +373,18 @@ defmodule Livebook.LiveMarkdown.MarkdownHelpersTest do
       assert markdown == reformat(markdown)
     end
 
+    test "surprise ordered list" do
+      markdown = "1986\\. What a great season."
+
+      assert markdown == reformat(markdown)
+    end
+
+    test "escaped Markdown" do
+      markdown = "not a \\[link\\]()"
+
+      assert markdown == reformat(markdown)
```

Comment:
Agree that the MathJax `\[` / `\]` delimiter pair is definitely not going to be usable in Markdown, those sequences are needed to keep things like \[this\]() away from the Markdown parser.

Going from HTML AST back to Markdown reliably seems to require a "Markdown escape" function, since the `[` and `]` appear unescaped in the HTML.